baseplate/lib/thrift_pool.py

```diff
@@ -203,10 +203,14 @@ def _is_stale(self, prot: TProtocolBase) -> bool:
         return False
 
     def _release(self, prot: Optional[TProtocolBase]) -> None:
-        if prot and prot.trans.isOpen():
-            self.pool.put(prot)
-        else:
+        try:
+            if prot and prot.trans.isOpen():
+                self.pool.put(prot)
+            else:
+                self.pool.put(None)
+        except:  # noqa: E722
             self.pool.put(None)
+            raise
```

Comment:
How about something like this?

```suggestion
        to_return = None
        try:
            if prot and prot.trans.isOpen():
                to_return = prot
        finally:
            self.pool.put(to_return)
```

I'm a little worried we'd accidentally double-release in some situations if we just put an extra `put()` in the error path.

Reply:
Yep that looks better.